### PR TITLE
TableReporter with no connectees no longer causes exception.

### PR DESCRIPTION
**Pull request by chrisdembia**
Fixes issue #1770 

### Brief summary of changes

This PR allows one to have a TableReporter that has no connectees. If this is the case, a warning message is emitted.

I found a bug in Simbody, which has been fixed via https://github.com/simbody/simbody/pull/565. However, I am not relying on this fix in this PR (I used a simple workaround with `resize()`), since I do not want to force other members of the dev team to rebuild Simbody at this time.

Lastly, I updated a few of the DataTable exceptions to provide more helpful messages, especially in MATLAB.

### Testing I've completed

Implemented a test in testComponentInterface.

### CHANGELOG.md

- no need to update because TableReporter has yet to be released.

The Doxygen for this PR can be viewed at http://myosin.sourceforge.net/?C=N;O=D; click the folder whose name is this PR's number.

### CODE REVIEW

File: OpenSim/Common/Reporter.h [hunk 1]
```diff
@@ -228,7 +234,14 @@ OpenSim_DECLARE_CONCRETE_OBJECT_T(TableReporter_, InputT, Reporter<InputT>);
...
             labels.push_back( input.getLabel(idx) );
         }
-        const_cast<Self*>(this)->_outputTable.setColumnLabels(labels);
+        if (!labels.empty()) {
+            const_cast<Self*>(this)->_outputTable.setColumnLabels(labels);
+        } else {
+            std::cout << "Warning: No outputs were connected to '"
+                      << this->getName() << "' of type "
+                      << getConcreteClassName()
+                      << "; did you intend to report any outputs?" << std::endl;
```
Comment (tkuchida) [inline OpenSim/Common/Reporter.h]: The last piece of the message doesn't seem to add anything. Perhaps tell the user "You can connect outputs by calling `addToReport()`"?

Comment (chrisdembia) [inline OpenSim/Common/Reporter.h]: Good idea; done.

Comment (tkuchida) [inline OpenSim/Common/Reporter.h]: Looks great :+1:

**Comment by chrisdembia** [comment]
I merged with master, which should fix the travis failure.

**Comment by carmichaelong** [comment]
Thanks for fixing the travis failure. In general LGTM. At first I was hesitant about throwing exceptions when printing an empty table (rather than just a warning), but 1) I think it's easy to take a wait and see approach; and 2) a change to that doesn't belong in this PR. I will merge now.

**Comment by chrisdembia** [comment]
Thanks @carmichaelong. I agree that toString() shouldn't throw the EmptyTable exception. I've created a separate issue for that.

**Pull request merged**